R/create_round.R

```diff
@@ -276,3 +279,61 @@ get_schema_round <- function(schema) {
     "items", "properties"
   )
 }
+# Check round_id pattern in `round_id` var values when
+# round_id_from_variable = TRUE
+check_round_id_pattern_vals <- function(model_tasks, round_id,
+                                        call = rlang::caller_env()) {
+  invalid_round_id_vals <- purrr::map(
+    model_tasks$model_tasks,
+    ~ {
+      round_id_var_vals <- purrr::pluck(
+        .x, "task_ids", round_id
+      )
+      invalid_round_id_var_patterns(round_id_var_vals) |>
+        purrr::compact()
+    }
+  )
+  invalid_round_id_vals <- purrr::set_names(
+    invalid_round_id_vals,
+    seq_along(invalid_round_id_vals)
+  )
+
+  if (length(unlist(invalid_round_id_vals)) > 0L) {
```

Comment:
when checking `round_id` variable from a variable, we report a bullet for each failure in `optional` and `required` properties and each modeling task separately.

Reply:
The github comments you've added here and throughout are really helpful! It could be helpful to include these annotations as comments in the code itself.